pyquil/paulis.py

```diff
@@ -78,10 +78,12 @@ def id(self):
             self._id = s
             return s
 
+    def __eq__(self, other):
+        return self.id() == other.id()
+
     def __len__(self):
-        """l
-        The length of the pauliterm is the number of Pauli operators in the term. This is
-        equivalent to the length of self._ops dict
+        """
+        The length of the pauliterm is the number of Pauli operators in the term.
```

Comment:
I'm not sure if this is what you intended, but this gives `len(sI(1))==0`, I think.